src/files/compose/validateDappnodeCompose.ts

```diff
@@ -0,0 +1,273 @@
+ */
+function validateComposeServiceNetworks(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string
+): void {
+  const cpService = compose.services[cpServiceName];
+  const cpServiceNetworks = cpService.networks;
+  if (!cpServiceNetworks) return;
+
+  for (const cpServiceNetwork of cpServiceNetworks) {
+    if (!cpServiceNetwork) continue;
+
+    if (typeof cpServiceNetwork === "string") {
+      // Check docker network is whitelisted when defined in array format
+      if (!params.DOCKER_WHITELIST_NETWORKS.includes(cpServiceNetwork))
+        aggregatedError.errors.push(
+          Error(
+            `Compose service ${cpServiceName} has a non-whitelisted docker network. Only docker networks ${params.DOCKER_WHITELIST_NETWORKS.join(
+              ","
+            )} are allowed`
+          )
+        );
+    } else {
+      const cpServiceNetworkObjectName = Object.keys(cpServiceNetwork)[0];
+      if (!cpServiceNetworkObjectName) continue;
+
+      // Check docker network is whitelisted when defined in object format
+      if (
+        !params.DOCKER_WHITELIST_NETWORKS.includes(cpServiceNetworkObjectName)
+      )
+        aggregatedError.errors.push(
+          Error(
+            `Compose service ${cpServiceName} has a non-whitelisted docker network: ${cpServiceNetworkObjectName}. Only docker networks ${params.DOCKER_WHITELIST_NETWORKS.join(
+              ","
+            )} are allowed`
+          )
+        );
+
+      // Check core aliases are not used by non core packages
+      const cpServiceNetworkAliases =
+        cpServiceNetwork[cpServiceNetworkObjectName].aliases;
+      if (
+        !isCore &&
+        cpServiceNetworkAliases &&
+        params.DOCKER_CORE_ALIASES.some(coreAlias =>
+          cpServiceNetworkAliases.includes(coreAlias)
+        )
+      ) {
+        aggregatedError.errors.push(
+          Error(
+            `Compose service ${cpServiceName} has a reserved docker alias. Aliases ${params.DOCKER_CORE_ALIASES.join(
+              ","
+            )} are reserved to core packages`
+          )
+        );
+      }
+    }
+  }
+}
+
+/**
+ * Ensure only core packages can use bind-mounted volumes
+ */
+function validateComposeAndComposeServiceVolumes(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string,
+  cpServiceVolume: string
+): void {
+  // Check that compose service volumes are defined also at top compose level
+  const cpVolumes = compose.volumes;
+  if (!cpVolumes) {
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has the volume ${cpServiceName} not defined at the top level. All docker volumes defined at the service level must be defined also at the top compose level volumes`
+      )
+    );
+    // return due to not having any volumes to check
+    return;
+  }
+
+  // Check that all volumes have names
+  const cpServiceVolumeName = cpServiceVolume.split(":")[0];
+  if (!cpServiceVolumeName)
```

Comment:
This check is not doing what it says. Both named volume mounts and bind mounts can use the `:` symbol. Please check the syntax in detail https://docs.docker.com/storage/volumes/

Also beware of long form volume definitions

Reply:
yes, but the main difference is:
- When there is a volume defined at the top level of the compose, it is a named volume
```
services:
  some: 
    volumes:
      - "some_volume:/some/path"
volumes:
  some_volume: {}
```

- When there is a volume only defined at the service level it is a bind mount volume
```
services:
  some: 
    volumes:
      - "/host/path:/some/path"
```


So the check for no bind mount volumes should be that every service volume must be defined at the top level of the compose

Reply:
Now it was simplified the function `validateComposeAndComposeServiceVolume`. It just ensures that the `volumeName` given exists at the top level of the compose file, this will ensure it is a named volume and not a bind mounted volume

see https://github.com/dappnode/DAppNodeSDK/blob/1561bee22808fc4d3b42d2b31c436e3ef7c45be7/src/files/compose/validateDappnodeCompose.ts#L218

Reply:
- See @3alpha reasoning https://github.com/dappnode/DNP_DAPPMANAGER/pull/1039

Reply:
that way uses another function under the hoods, I consider this way purer and cleaner